saleor/attribute/models/page.py

```diff
@@ -66,3 +79,17 @@ class Meta:
 
     def get_ordering_queryset(self):
         return self.page_type.attributepage.all()
+
+
+@receiver(m2m_changed, sender=AssignedPageAttribute.values.through)
+def handle_values_changed(sender, instance, action, **kwargs):
+    if action == "post_add":
+        # Get the newly added AttributeValues
+        added_values = kwargs.get("pk_set", [])
+
+        for value_id in added_values:
+            # Find the corresponding AssignedPageAttributeValue instance
+            apa_value = AssignedPageAttributeValue.objects.get(
+                value_id=value_id, assignment=instance
+            )
+            apa_value.save()
```

Comment:
This code executes 2 database queries for each element.

Reply:
I removed and change the approach for more efficient solution.